### PR TITLE
Add matchtype 'none'

### PR DESCRIPTION
**Pull request by frankforpresident**
While testing the plugin I've notice that the default behaviour was not working anymore.

I've testing it locally and fixed the bug

### CODE REVIEW

**Comment by frankforpresident** [comment]
I didn't understand why this was introduced while the tests were passing but I had another look at unit tests and I've noticed that the negative tests will never work like this because it will always be rejected when one header value turns 403. 

I would recommend to update the negative tests to validate every header value one by one and see if it will return 403 independently. Currently if one value returns 200 it will not be catched by TestHeadersNotMatch

**Comment by frankforpresident** [comment]
I've added an extra unit test to prevent this bug in the future.

**Comment by frankforpresident** [comment]
@dkijkuit,

I've added "matchtype none" which provides me with enough flexibility to no use the regexp2 package. 

All tests passed and I've tested it locally

**Comment by frankforpresident** [comment]
@dkijkuit,

Could you please have a look?

File: README.MD [hunk 1]
```diff
@@ -82,7 +82,7 @@ http:
               name: "HEADER_4"
               matchtype: one
               values:
-                - "VALUE_\\d"
+                - "VALUE_\d"
```
Comment (dkijkuit) [inline README.MD:85]: This should be escaped as well

File: README.MD [hunk 2]
```diff
@@ -157,7 +157,7 @@ middlewares:
...
```
Comment (dkijkuit) [inline README.MD:160]: Escape

File: dynamic-dev-config.yaml [hunk 3]
```diff
@@ -50,6 +50,6 @@ http:
               name: "HEADER_4"
               matchtype: one
               values: 
-                - "VALUE_\\d"
+                - "VALUE_\d"
```
Comment (dkijkuit) [inline dynamic-dev-config.yaml:53]: Escape

**Review by dkijkuit — CHANGES_REQUESTED** [review]
Please make sure the docs are correct, ie. escaping the regex

File: dynamic-dev-config.yaml (hunk 3 above)
Comment (frankforpresident) [reply dynamic-dev-config.yaml]: done

**Comment by frankforpresident** [comment]
> Please make sure the docs are correct, ie. escaping the regex

Done

**Review by dkijkuit — APPROVED** [review]
